pkg/common/oras/authprovider/aws/awsecrbasic.go

```diff
@@ -55,6 +56,9 @@ func getEcrAuthToken() (EcrAuthToken, error) {
 	region := os.Getenv("AWS_REGION")
 	roleArn := os.Getenv("AWS_ROLE_ARN")
 	tokenFilePath := os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE")
+	apiOverrideEndpoint := os.Getenv("AWS_API_OVERRIDE_ENDPOINT")
```

Comment:
just curious, do we need to add existence checks for these fields?

Reply:
@akashsinghal To be clear, I was adding checks to the helm chart deployment template; if any one field was missing or empty, all fields would be excluded. The logic in the code will revert to the default resolver, if a custom resolver cannot be resolved.

Reply:
I pulled and rebased, then pushed a signed commit. We still need to resolve the conflict, listed below.

Reply:
Thanks @jimmyraywv! I will resolve the conflict on your behalf

Reply:
> @akashsinghal To be clear, I was adding checks to the helm chart deployment template; if any one field was missing or empty, all fields would be excluded. The logic in the code will revert to the default resolver, if a custom resolver cannot be resolved.

I see that also works. Our typical code pattern has been to do all environment variable existence checks in the code. (just in case we are running this in a non k8 environment). But I don't think this is a blocker since we are only targeting this for k8 :)